argo/argo-build-workflow.yaml

```
@@ -0,0 +1,218 @@
+        templateRef:
+          name: workflow-template-common-template-1.0
+          template: get-src
+        arguments:
+          parameters:
+          - name: repo
+            value: "{{workflow.parameters.repo}}"
+          - name: branch
+            value: "{{workflow.parameters.branch}}"
+
+      - name: go-build
+        dependencies: [dependencies]
+        template: go-build
+        arguments:
+          parameters:
+          - name: notifiable_commits
+            value: "{{tasks.dependencies.outputs.parameters.notifiable_commits}}"
+          artifacts:
+          - name: src
+            from: "{{tasks.dependencies.outputs.artifacts.src}}"
+
+      - name: go-tests
+        dependencies: [go-build]
+        templateRef:
+          name: workflow-template-common-template-1.0
+          template: go-tests
+        arguments:
+          parameters:
+          - name: notifiable_commits
+            value: "{{tasks.dependencies.outputs.parameters.notifiable_commits}}"
+          - name: repo
+            value: "{{workflow.parameters.repo}}"
+          - name: test_path
+            value: .
+        when: "{{workflow.parameters.skip_tests}} == false"
+
+      - name: sonar-branch-analysis
+        dependencies: [go-tests]
+        templateRef:
+          name: workflow-template-sonar-template-1.0
+          template: sonar-branch-analysis
+        arguments:
+          parameters:
+          - name: sonar_projectKey
+            value: "{{workflow.parameters.repo}}"
+          - name: sonar_sources
+            value: .
+          - name: sonar_tests
+            value: .
+          - name: sonar_branch_name
+            value: "{{workflow.parameters.branch}}"
+          artifacts:
+          - name: coverage
+            from: "{{tasks.go-tests.outputs.artifacts.coverage}}"
+            optional: true
+        when: "!('{{workflow.parameters.pull_request}}' =~ '^[0-9]+$')"
+
+      - name: sonar-pr-analysis
+        dependencies: [go-tests]
+        templateRef:
+          name: workflow-template-sonar-template-1.0
+          template: sonar-pr-analysis
+        arguments:
+          parameters:
+          - name: sonar_projectKey
+            value: "{{workflow.parameters.repo}}"
+          - name: sonar_sources
+            value: .
+          - name: sonar_tests
+            value: .
+          - name: sonar_pullrequest_key
+            value: "{{workflow.parameters.pull_request}}"
+          - name: sonar_pullrequest_branch
+            value: "{{workflow.parameters.branch}}"
+          - name: sonar_pullrequest_base
+            value: "{{workflow.parameters.pull_request_base}}"
+          artifacts:
+          - name: coverage
+            from: "{{tasks.go-tests.outputs.artifacts.coverage}}"
+            optional: true
+        when: "('{{workflow.parameters.pull_request}}' =~ '^[0-9]+$')"
+
+
+  - name: go-build
+    inputs:
+      parameters:
+      - name: go-image
+        value: golang:1.14.0
```

We have a new patch for the language available
```suggestion
        value: golang:1.14.4
```